Python/optimizer.c

```diff
@@ -185,12 +185,14 @@ _PyOptimizer_Optimize(
     else {
         executor->vm_data.code = NULL;
     }
+    executor->vm_data.chain_depth = chain_depth;
+    assert(executor->vm_data.valid);
     _PyExitData *exit = _tstate->jit_tracer_state.initial_state.exit;
     if (exit != NULL) {
         exit->executor = executor;
+    } else {
+        Py_DECREF(executor);
```

Comment:
Could you please add a comment saying:
An executor inserted into the code object now has a strong reference to it from the code object. Thus, we don't need this reference anymore.